Add people breakdown on retention

EDsCODE opened the pull request:
## Changes
- adds ability to click and breakdown who is retaining


Todo:
- [ ] order by people who go through flow
- [ ] paginate
- [ ] add people tests

## Checklist

- [ ] All querysets/queries filter by Team (if applicable)
- [ ] Backend tests (if applicable)
- [ ] Cypress E2E tests (if applicable)

EDsCODE committed:
use pandas to intercept unique ids

EDsCODE committed:
add people breakdown

EDsCODE committed:
update test

EDsCODE committed:
add people

EDsCODE committed:
remove people tests

EDsCODE committed:
ranked people

EDsCODE committed:
fixed filtering

EDsCODE committed:
merged fix

EDsCODE committed:
people working again

EDsCODE committed:
people paginate on retention working

EDsCODE committed:
fix merge conflict

EDsCODE committed:
fixed sorting

EDsCODE committed:
fix types

EDsCODE committed:
.

EDsCODE committed:
merge master and fix conflicts

timgl reviewed (commented):
Mostly looks good, however the error state really confused me here:
![image](https://user-images.githubusercontent.com/1727427/87795505-725b2d80-c848-11ea-8fb5-4fa11a4e59ae.png)

EDsCODE committed:
add message for no users

EDsCODE commented:
I added a message when there are no users. Also, in the date filtering PR, I changed the percentage to show 0 when there's no one rather than 100 so it will be less confusing

<img width="588" alt="Screen Shot 2020-07-17 at 10 22 44 AM" src="https://user-images.githubusercontent.com/13127476/87796605-93eff100-c817-11ea-839c-a02dade3e8d7.png">

timgl commented on frontend/src/scenes/retention/RetentionTable.js:
```diff
@@ -46,7 +54,107 @@ export function RetentionTable({ logic }) {
+                    {retention && !peopleLoading ? (
+                        <div>
+                            {retention?.data[selectedRow]?.values[0]?.count === 0 ? (
+                                <span>No Users during this period</span>
```
Nit but we don't capitalise users anywhere else I think

EDsCODE committed:
make lowercase

Twixes commented on frontend/src/scenes/retention/retentionTableLogic.js:
```diff
@@ -24,6 +46,21 @@ export const retentionTableLogic = kea({
+        },
+        retention: {
+            updateRetention: (_, { retention }) => retention,
+        },
+        loadingMore: [
+            false,
+            {
+                updateLoadingMore: (state, { status }) => status,
+            },
+        ],
```
Isn't this pattern (with `actions.updateLoadingMore`) kind of a Kea no-no? https://kea.js.org/docs/guide/concepts#listeners

EDsCODE replied:
yup good point. it's unnecessary

EDsCODE committed:
remove unnecessary action

EDsCODE committed:
make sure something is always returned

Twixes committed:
Add period and update button capitalization
